packages/core/deprecations/core-deprecations-browser-internal/src/deprecations_client.ts

```diff
@@ -8,16 +8,17 @@
 
 import { i18n } from '@kbn/i18n';
 import type { HttpStart } from '@kbn/core-http-browser';
-import type { DomainDeprecationDetails, DeprecationsGetResponse } from '../../server/types';
+import type {
+  DomainDeprecationDetails,
+  DeprecationsGetResponse,
+} from '@kbn/core-deprecations-common';
+import type { ResolveDeprecationResponse } from '@kbn/core-deprecations-browser';
 
 /* @internal */
 export interface DeprecationsClientDeps {
   http: Pick<HttpStart, 'fetch'>;
 }
 
-/* @internal */
-export type ResolveDeprecationResponse = { status: 'ok' } | { status: 'fail'; reason: string };
-
 export class DeprecationsClient {
```

Comment:
I wonder if this should also be annotated with the @internal annotation. Don't know the exact policy.

Reply:
If anything, it should actually be public since [upgrade_assistant](https://github.com/elastic/kibana/blob/main/x-pack/plugins/upgrade_assistant/server/lib/kibana_status.ts#L8) imports and uses the DeprecationsClient from core.